src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -171,7 +172,23 @@ def _(self, node_list: list) -> list[QASMNode]:
 
     @visit.register
     def _(self, node: Program) -> None:
-        self.visit(node.statements)
+        i = 0
+        while i < len(node.statements):
+            stmt = node.statements[i]
```

Comment:
Probably easier to just enumerate the nodes
```suggestion
       for i, stmt in enumerate(node.statements):
```
and inside the statement simply check `node.statements[i + 1]` if `stmt` is a verbatim pragma.